src/App/App.tsx

```diff
@@ -71,14 +74,47 @@ library.add(faExternalLinkSquareAlt);
 toast.configure({
   autoClose: TOAST_AUTO_CLOSE_DELAY /** defines how long a toast remains visible on screen */,
 });
-initGoogleAnalytics();
+
+const initializeOptions = {
+  testMode: GA_ENV_TEST === GA_ENV,
+};
+
+/** Initialize google analytics */
+initGoogleAnalytics(GA_CODE, initializeOptions);
 
 import store from '../store';
 import { getOauthProviderState } from '../store/selectors';
 import './App.css';
 
+/** Interface defining component state */
+export interface AppState {
+  username?: string;
+}
+
 /** Main App component */
-class App extends Component {
+class App extends Component<{}, AppState> {
```

Comment:
@kelvin-muchiri @moshthepitt @p-netm - how do you guys feel about converting this to a functional component? Is that within the scope of this PR or should/could it be a part of a different ticket?

Reply:
@cKellyDesign lets move it to a different issue.  But I like the idea!

Reply:
https://github.com/onaio/reveal-frontend/issues/756